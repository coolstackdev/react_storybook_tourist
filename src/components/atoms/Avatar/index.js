import React from 'react'
import T from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div``

const Image = styled.img`
  border-radius: 50%;
  object-fit: cover;
  width: 4rem;
  height: 4rem;
  display: inline-block;
  position: relative;
  z-index: 1;
  user-select: none;
`

const Avatar = ({ avatarUrl, className }) => {
  return (
    <Wrapper className={className}>
      <Image src={avatarUrl} />
    </Wrapper>
  )
}

Avatar.propTypes = {
  avatarUrl: T.string,
}

export default Avatar
