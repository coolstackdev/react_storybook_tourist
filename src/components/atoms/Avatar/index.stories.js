import React from 'react'
import { storiesOf } from '@storybook/react'
import Avatar from '.'

storiesOf('Avatar', module)
  .add('default', () => (
    <Avatar avatarUrl="https://i.ytimg.com/vi/vOWboQGcm88/maxresdefault.jpg" />
  ))
  .add('reverse', () => (
    <Avatar reverse avatarUrl="https://i.ytimg.com/vi/vOWboQGcm88/maxresdefault.jpg" />
  ))
