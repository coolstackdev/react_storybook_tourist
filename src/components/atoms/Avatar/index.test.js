import Avatar from '.'

// const wrap = (props = {}) => shallow(<Avatar {...props} />)

describe('Avatar', () => {
	let wrapper

	describe('when Avatar renders', () => {
		beforeEach(() => {
			wrapper = shallow(
				<Avatar avatarUrl="https://i.ytimg.com/vi/vOWboQGcm88/maxresdefault.jpg" />
			)
		})

		it('works', () => {
			expect(
				wrapper
			).toMatchSnapshot()
		})

		it('should render when an image is passed as props', () => {
			expect(
				wrapper.exists()
			).toBe(true)
		})

		// it('props.avatarUrl should be put into img src', () => {
		// 	const img = wrapper.find('img')
		// 	expect(
		// 		img.props().src
		// 	).toBe("https://i.ytimg.com/vi/vOWboQGcm88/maxresdefault.jpg")
		// })
	})
})