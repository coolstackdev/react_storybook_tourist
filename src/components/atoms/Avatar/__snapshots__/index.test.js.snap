// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar when Avatar renders works 1`] = `
<Avatar__Wrapper>
  <Avatar__Image
    src="https://i.ytimg.com/vi/vOWboQGcm88/maxresdefault.jpg"
  />
</Avatar__Wrapper>
`;
